### PR TITLE
Feature/alcs 981 - NOI step 6 ROSO

### PR DESCRIPTION
**Pull request by mhuseinov**
- step 6 for ROSO
- unit tests

### CODE REVIEW

File: portal-frontend/src/app/features/notice-of-intents/edit-submission/additional-information/roso/delete-structure-confirmation-dialog/delete-structure-confirmation-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  <h4>Delete Structure Type</h4>
+</div>
+
+<div mat-dialog-content class="content">
+  <app-warning-banner>
+    <b>Warning:</b> Deleting the structure type can remove some content already saved to this page. <br />
+    Do you want to continue?
+  </app-warning-banner>
+  <div class="controls">
+    <button mat-stroked-button color="primary" (click)="onCancel()">cancel</button>
```
Comment (dhaselhan) [inline portal-frontend/src/app/features/notice-of-intents/edit-submission/additional-information/roso/delete-structure-confirmation-dialog/delete-structure-confirmation-dialog.component.html:11]: You can replace onCancel and onConfirm with [mat-dialog-close]="false" and [mat-dialog-close]="true" to achieve the same results.

File: services/apps/alcs/src/portal/notice-of-intent-submission/notice-of-intent-submission.entity.ts [hunk 2]
```diff
@@ -15,6 +15,11 @@ import { User } from '../../user/user.entity';
 import { ColumnNumericTransformer } from '../../utils/column-numeric-transform';
 import { NoticeOfIntentOwner } from './notice-of-intent-owner/notice-of-intent-owner.entity';
 
+export class ProposedStructure {
+  type: 'Lot' | 'Road Dedication' | null;
```
Comment (dhaselhan) [inline services/apps/alcs/src/portal/notice-of-intent-submission/notice-of-intent-submission.entity.ts:19]: Is this typing correct? I think the front end has a lot more types.

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

File: portal-frontend/src/app/features/notice-of-intents/edit-submission/additional-information/roso/delete-structure-confirmation-dialog/delete-structure-confirmation-dialog.component.html (hunk 1 above)
Comment (mhuseinov) [reply portal-frontend/src/app/features/notice-of-intents/edit-submission/additional-information/roso/delete-structure-confirmation-dialog/delete-structure-confirmation-dialog.component.html]: done

**Pull request merged**